Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -581,27 +597,32 @@ void SourceBufferPrivate::didReceiveInitializationSegment(InitializationSegment&
 {
     processPendingMediaSamples();
 
-    m_currentAppendProcessing = m_currentAppendProcessing->whenSettled(RunLoop::current(), [segment = WTFMove(segment), weakThis = WeakPtr { *this }, this](auto&& result) mutable -> Ref<MediaPromise> {
+    auto segmentCopy = segment;
+    m_currentAppendProcessing = m_currentAppendProcessing->whenSettled(RunLoop::current(), [segment = WTFMove(segment), weakThis = WeakPtr { *this }, this, abortCount = m_abortCount](auto&& result) mutable {
```

Comment:
Do we need segmentCopy if we just copy segment instead of moving it in this lambda?

Reply:
It was a matter of having just 1 copy vs 2.

Reply:
oh I see what you mean.

The issue is in the order of running of the capture. 
the segment needs to be copied and then moved later.
but when doing 
`blah([obj = WTFMove(obj] { })->blah2([obj = WTFMove(obj) { } );`

where is the move happening first?
C++17 defines that when you do F(A), F is evaluated before A. But at a glance, I find it super dodgy. 
so in theory:
`blah([obj] { })->blah2([obj = WTFMove(obj) { } );` should do what we want.
But I find it much clearer to explicitly take the copy first and move that copy.